src/main/java/org/breedinginsight/brapi/v2/services/BrAPIGermplasmService.java

```diff
@@ -82,10 +82,15 @@ public Optional<BrAPIGermplasm> getGermplasmByDBID(UUID programId, String germpl
         return germplasmDAO.getGermplasmByDBID(germplasmId, programId);
     }
 
-    public List<Map<String, Object>> processListData(List<BrAPIGermplasm> germplasm, BrAPIListDetails germplasmList){
+    public List<Map<String, Object>> processListData(List<BrAPIGermplasm> germplasm, BrAPIListDetails germplasmList, Program program){
         Map<String, BrAPIGermplasm> germplasmByName = new HashMap<>();
         for (BrAPIGermplasm g: germplasm) {
-            germplasmByName.put(g.getGermplasmName(), g);
+            // Use the full, unique germplasmName with programKey and accessionNumber (GID) for 2 reasons:
+            // 1. the BrAPI list items are full names, and
+            // 2. germplasmNames alone are not unique within a program, this led to unexpected behavior, see BI-2344.
+            String uniqueGermplasmName = String.format("%s [%s-%s]", g.getGermplasmName(), program.getKey(), g.getAccessionNumber());
+            g.setGermplasmName(uniqueGermplasmName);  // For later.
```

Comment:
could you clarify what the "for later" comment means?

Reply:
I updated the comment to be more clear. Inside that loop, `g` is an element of the `germplasm` collection which is used again on line 111, and needs to have the full germplasmName with programKey and accession number later in the method.